scripts/chopsticks-test.ts

```diff
@@ -0,0 +1,139 @@
+
+    const destAddress = getRandomAddress(destAdapter.getSS58Prefix());
+    
+    let min = await firstValueFrom(originAdapter.subscribeMinInput(token, toChain));
+    // don't use console.log because I don't want newline here - I want the OK/FAIL to be added on the same line
+    process.stdout.write(`Transferring ${min} ${token} from ${fromChain} to ${toChain}.${destAddress}... `);
+
+    let expectedDiff = FN.fromInner("1", min.getPrecision());
+
+    const tx = originAdapter.createTx({
+      to: toChain,
+      token,
+      amount: min,
+      address: destAddress,
+      signer: destAddress, // doesn't matter in chopsticks
+    });
+
+    let cfg = {
+        token: token,
+        address: destAddress,
+        amount: expectedDiff,
+        timeout: 15*1000, // 15 sec timeout
+    };
+    let changes = destAdapter.subscribeBalanceChanged(cfg);
+    await submitTx(tx as SubmittableExtrinsic<"rxjs", ISubmittableResult>);
+    let q = changes.pipe(takeWhile((x) => x == BalanceChangedStatus.CHECKING, true));
+
+    let result = await lastValueFrom(q);
+    let success = result == BalanceChangedStatus.SUCCESS;
+    if (success) {
+        console.log("OK");
+    } else {
+        console.log("FAIL");
+    }
+    return success;
+}
+
+async function main(): Promise<void> {
+
+    const availableAdapters: Record<string, BaseCrossChainAdapter> = {
+        kusama: new KusamaAdapter(),
+        statemine: new StatemineAdapter(),
+        kintsugi: new KintsugiAdapter(),
+        karura: new KaruraAdapter(),
+        heiko: new HeikoAdapter(),
+    };
+
+    const bridge = new Bridge({
+        adapters: Object.values(availableAdapters),
+    });
+    const chains = Object.keys(availableAdapters) as ChainName[];
+
+    const provider = new ApiProvider("mainnet"); // we overwrite endpoints, so not really mainnet
+    const endpoints = {
+        kusama: ['ws://127.0.0.1:8004'],
+        kintsugi: ['ws://127.0.0.1:8000'],
+        statemine: ['ws://127.0.0.1:8001'],
+        karura: ['ws://127.0.0.1:8002'],
+        heiko: ['ws://127.0.0.1:8003'],
+    };
+
+    // connect all adapters
+    // const connected = 
+    await lastValueFrom(
+        provider.connectFromChain(chains, endpoints)
+    );
+    // and set apiProvider for each adapter
+    await Promise.all(
+        chains.map((chain) =>
+            availableAdapters[chain].setApi(provider.getApi(chain))
+        )
+    );
+
+    let testcases = [
+        ["karura", "KBTC"],
+        ["karura", "KINT"],
+        ["karura", "LKSM"],
+        ["kusama", "KSM"],
+        ["statemine", "USDT"],
+        ["heiko", "KINT"],
+        ["heiko", "KBTC"],
+    ].flatMap(([to, token]) => [["kintsugi", to, token], [to, "kintsugi", token]]); // bidirectional testing
+
+    for (const [from, to, token] of testcases) {
+        if (!await checkTransfer(from as ChainName, to as ChainName, token, bridge)) {
+            console.log("Exiting unsuccessfully");
+            process.exit(-1);
```

Comment:
Would it make sense to allow all tx attempts to run, and just collect whether any of them failed instead of aborting on first failure?

Reply:
Yea I had already done that, just forgot to push..